python/educational-core/student/src/com/jetbrains/edu/learning/stepic/EduStepicConnector.java

```diff
@@ -55,30 +56,59 @@
 import java.security.cert.X509Certificate;
 import java.util.*;
 
+@Deprecated
 public class EduStepicConnector {
   private static final Logger LOG = Logger.getInstance(EduStepicConnector.class.getName());
   private static final String stepicUrl = "https://stepic.org/";
   private static String ourCSRFToken = "";
+  private static String accessToken = "";
   private static CloseableHttpClient ourClient;
+  private static final String CLIENT_ID = "hUCWcq3hZHCmz0DKrDtwOWITLcYutzot7p4n59vU";
 
   //this prefix indicates that course can be opened by educational plugin
   public static final String PYCHARM_PREFIX = "pycharm";
+  public static final String CODE_PREFIX = "code";
+  public static final String PYTHON27 = "python27";
+  public static final String PYTHON3 = "python3";
   private static BasicCookieStore ourCookieStore;
 
+  static final private Gson GSON =
+    new GsonBuilder().registerTypeAdapter(TaskFile.class, new StudySerializationUtils.Json.StepicTaskFileAdapter())
+      .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+
   private EduStepicConnector() {
   }
 
+  // TODO : merge. look at comments
   public static StepicUser login(@NotNull final String username, @NotNull final String password) {
     initializeClient();
-    if (postCredentials(username, password)) {
-      final StepicWrappers.AuthorWrapper stepicUserWrapper = getCurrentUser();
-      if (stepicUserWrapper != null && stepicUserWrapper.users.size() == 1) {
-        return stepicUserWrapper.users.get(0);
-      }
+    StepicWrappers.TokenInfo tokenInfo = postCredentials(username, password);
```

Comment:
Do not see the reason to return token info here instead of boolean success/failure. (and assign accessToken = tokenInfo.accessToken; in postCredentials method)